### PR TITLE
Add geek details

### PR DESCRIPTION
**Pull request by lxpollitt**

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 03902362058ec8f29ecba2eeba2bf2676a03594e

https://deploy-preview-3229--calico.netlify.com

File: _includes/geek-details/datastore-etcd.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Calico stores the operational and configuration state of your cluster in a central datastore. If the datastore is unavailable your Calico network continues operating, but cannot be updated (no new pods can be networked, no policy changes can be applied, etc.).
```
Comment (ozdanborne) [inline _includes/geek-details/datastore-etcd.md]: Is this geek-detail used anywhere? I only see datastore-kubernetes being used on the GKE page

File: _includes/geek-details.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            <script>
+                $(document).ready(function(){
+                $("#geekValue{{ kv[0] }}").click(function(){
+                    $("#geekModal{{ kv[0] }}").modal();
+                });
+                });
+            </script>    
+        {% endfor %}
+    </div>
+    <button class="geek-details-info" id="geekDetailsInfo">?</button>
```
Comment (ozdanborne) [inline _includes/geek-details.html]: When I click the question mark, I'm getting details about "Kubernetes Datastore". Is that the intent? I'm on chrome

Comment (ozdanborne) [reply _includes/geek-details.html]: Oh wait, I'm getting general info, but the title is "Kubernetes Datastore".

**Comment by ozdanborne** [comment]
@lxpollitt I'm not comfortable adding in all the blank placeholders `_includes` for geek details you're planning on adding later. Can you remove the ones that are unused?

File: _includes/geek-details/datastore-etcd.md (hunk 1 above)
Comment (lxpollitt) [reply _includes/geek-details/datastore-etcd.md]: It's not used by GKE page, but was easy to author while doing the kdd one.  All of the install pages will have geek details once we are finished.  In addition we anticipate using geek details on some of the how-to or reference pages in a "this requires the following" type of style.

File: _includes/geek-details.html (hunk 2 above)
Comment (lxpollitt) [reply _includes/geek-details.html]: Good catch - will fix.

**Comment by lxpollitt** [comment]
> @lxpollitt I'm not comfortable adding in all the blank placeholders `_includes` for geek details you're planning on adding later. Can you remove the ones that are unused?

Can you say more about why you are not comfortable?  They define the valid/expected combinations of options that we will need - so should reduce chance of us messing this up as we fill in the content for them over the next week.

File: _includes/geek-details.html (hunk 2 above)
Comment (lxpollitt) [reply _includes/geek-details.html]: Fixed

**Comment by lxpollitt** [comment]
@ozdanborne Ready for another review.  I will continue to improve descriptions and use of the widget in more places over the next week or so, but would like to get this first PR merged as a base if possible.

File: getting-started/kubernetes/flannel/flannel.md [hunk 3]
```diff
@@ -79,3 +82,6 @@ etcd, complete the following steps.
 
 1. If you wish to enforce application layer policies and secure workload-to-workload
    communications with mutual TLS authentication, continue to [Enable application layer policy]({{site.baseurl}}/security/app-layer-policy) (optional).
+
+{% include geek-details.html details='Policy:Calico,IPAM:Host-local,CNI:Calico,Overlay:VXLAN,Routing:Static,Datastore:etcd' %}
```
Comment (ozdanborne) [inline getting-started/kubernetes/flannel/flannel.md]: Did you intentionally put this second one at the bottom of the flannel page? Seems odd to me.

Comment (lxpollitt) [reply getting-started/kubernetes/flannel/flannel.md]: It's meant to be there, but is missing the intro sentence "The geeky details of what you get:".  

I'll fix that up.

**Pull request merged**